### PR TITLE
[12.0] stock_move_sale_invoice_partner: add tests

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**

### CODE REVIEW

File: stock_move_sale_invoice_partner/tests/test_stock_move_sale_invoice_partner.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            {
+                "name": "Invoice Customer - test",
+                "type": "invoice",
+                "parent_id": self.partner.id,
+            }
+        )
+        self.invoice_partner1 = self.env["res.partner"].create(
+            {"name": "Real Invoice Customer", "parent_id": self.partner.id}
+        )
+        self.picking_type = self.PickingType.search(
```
Comment (yostashiro) [inline stock_move_sale_invoice_partner/tests/test_stock_move_sale_invoice_partner.py]:
```suggestion
        self.picking_type = self.env["stock.picking.type"].search(
```

File: stock_move_sale_invoice_partner/tests/test_stock_move_sale_invoice_partner.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
+from odoo.tests import tagged
+from odoo.tests.common import TransactionCase
+
+
+@tagged("post_install", "-at_install")
+class TestStockMoveSaleInvoicePartner(TransactionCase):
+    def setUp(self):
+        super(TestStockMoveSaleInvoicePartner, self).setUp()
+        self.advance_inv_model = self.env["sale.advance.payment.inv"]
+        self.PickingType = self.env["stock.picking.type"]
```
Comment (yostashiro) [inline stock_move_sale_invoice_partner/tests/test_stock_move_sale_invoice_partner.py]:
```suggestion
```

File: stock_move_sale_invoice_partner/tests/test_stock_move_sale_invoice_partner.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        move = self.StockMove.search([("picking_id", "=", picking.id)])
+        self.assertEqual(self.invoice_partner, move.invoice_partner_id)
+
+        wizard = self.advance_inv_model.create({"advance_payment_method": "delivered"})
+        wizard.with_context(
+            active_model="sale.order",
+            active_ids=[order.id],
+            active_id=order.id,
+        ).create_invoices()
+        invoice = order.mapped("invoice_ids")
```
Comment (yostashiro) [inline stock_move_sale_invoice_partner/tests/test_stock_move_sale_invoice_partner.py]:
```suggestion
        invoice = order.invoice_ids
```

File: stock_move_sale_invoice_partner/tests/test_stock_move_sale_invoice_partner.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        invoice.write({"partner_id": self.invoice_partner1.id})
+        self.assertEqual(self.invoice_partner1, move.invoice_partner_id)
```
Comment (yostashiro) [inline stock_move_sale_invoice_partner/tests/test_stock_move_sale_invoice_partner.py]:
```suggestion
        self.assertEqual(move.invoice_partner_id, self.invoice_partner1)
```

**Review by yostashiro — APPROVED** [review]
Non-blocking comments out of preferences.

File: stock_move_sale_invoice_partner/tests/test_stock_move_sale_invoice_partner.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+from odoo.tests.common import TransactionCase
+
+
+@tagged("post_install", "-at_install")
+class TestStockMoveSaleInvoicePartner(TransactionCase):
+    def setUp(self):
+        super(TestStockMoveSaleInvoicePartner, self).setUp()
+        self.advance_inv_model = self.env["sale.advance.payment.inv"]
+        self.StockMove = self.env["stock.move"]
+        self.Picking = self.env["stock.picking"]
```
Comment (kanda999) [inline stock_move_sale_invoice_partner/tests/test_stock_move_sale_invoice_partner.py:11]: we don't need StockMove or Picking as well as PickingType.

File: stock_move_sale_invoice_partner/tests/test_stock_move_sale_invoice_partner.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+                "type": "invoice",
+                "parent_id": self.partner.id,
+            }
+        )
+        self.invoice_partner1 = self.env["res.partner"].create(
+            {"name": "Real Invoice Customer", "parent_id": self.partner.id}
+        )
+        self.picking_type = self.env["stock.picking.type"].search(
+            [("warehouse_id", "=", self.warehouse.id), ("code", "=", "outgoing")]
+        )
```
Comment (kanda999) [inline stock_move_sale_invoice_partner/tests/test_stock_move_sale_invoice_partner.py:39]: What is this used for?

**Review by kanda999 — CHANGES_REQUESTED** [review]
I think it's a test you made a long ago, so if there's anything to fix please fix it.

**Pull request closed without merging**